### ISSUE TITLE
Repeating Webhook calls

### ISSUE THREAD

**Issue by Jwerfel** [ZekeSnider/Jared#73]
I've seen this a few times where when I send a message that uses a webhook, Jared will repeat making the call to the webhook and then sending me the response.  This only happens on messages that link to webhooks (other built in messages like /ping) only fire once.  I've confirmed Jared is making the call to the webhook (not just repeating the response).  It doesn't happen every time but the only way I can stop it is to restart Jared.  Any ideas why this might be happening?

Thank you

**Comment by ZekeSnider** [comment]
I've not seen this before, would be curious if you're able to narrow down reproduction steps. Are you responding to the webhook request with a message to send?

**Comment by Jwerfel** [comment]
Yes, I am responding to the webhook calls with a message to send (which is working).  On the extra messages, I noticed the sender is my iMessage account Jared is using (not the sender of the original request) and the message Jared is sending to the webhook is the response from the first webhook message.  So for some reason, Jared is receiving the first webhook response, sending it to me on iMessage and then taking that response and calling my webhook repeatedly.

My solution for the moment is on the webhook side look to see if sender.isMe = true and if so, I ignore it.  That has stopped the problem from repeating but obviously is not a perfect solution.

Are there any logs for Jared I should be looking at?  

Thank you

**Comment by ZekeSnider** [comment]
Ah gotcha. So I don't believe there is any logic to _not_ send webhook events for messages triggered from Jared in the first place. So that explains why you're getting events for the message you sent in the webhook response. It's a bit tricky since we don't get a message ID when we send a message. Maybe it could be de-duped by matching the text as an imperfect solution?

I'll mark this ticket as an enhancement, will update if I come up with a better solution.

**Comment by Skeyelab** [comment]
I experience this too.  I just filter the message in my webhook code.
